mqtt/mqtt-broker/tests/persist_failpoints.rs

```diff
@@ -90,7 +90,8 @@ fn test_failpoints_smoketest() {
             let mut persistor = FilePersistor::new(path, BincodeFormat::new());
 
             let result = persistor.load().await;
-            assert!(result.is_err());
+            let err = result.unwrap_err();
+            assert_eq!(&ErrorKind::TaskJoin, err.kind());
```

Comment:
`assert_match(result, Err(ErrorKind::TaskJoin.into()));` instead?

Reply:
I had considered that, but didn't want to bring in another dependency just for this one simple case.